Makefile

```diff
@@ -12,12 +13,15 @@ locale/downgrade.pot: downgrade
 		--from-code=utf-8 -L shell \
 		--package-name=downgrade \
 		--copyright-holder=$(AUTHOR) \
-		-o $@ $<
+		-o locale/downgrade.pot downgrade
+	ls ./locale/*po | xargs -I {} msgmerge --update {} ./locale/downgrade.pot
```

Comment:
This now means `locale/*po` should be added as dependencies to this target. If one of those change (but `downgrade` doesn't), we want this rule to fire.

Reply:
Can you replace `ls | xargs` with `find -exec`? See [ParsingLs](https://mywiki.wooledge.org/ParsingLs).

Or better, once `locale/*po` are listed pre-requisites, there is an automatic variable for them (`$^`):

```make
locale/downgrade.pot: downgrade locale/*po
        xgettext \
          -o $@ $<
        msgmerge --update $^ $@
```

One bother is that `$^` is all pre-requisites, so it includes `downgrade` in the list. Will that error? I wonder what the nicest way to chop that off would be :thinking:

Reply:
Changed all `ls` related pipes to `find` to address this. Also changed all target/dependency inputs to their special tokens.

> This now means `locale/*po` should be added as dependencies to this target. If one of those change (but `downgrade` doesn't), we want this rule to fire.

From my perspective, the typical workflow for updating `*po` files should be as follows:

1. Update `downgrade`
2. Run `make locale`, which will update `downgrade.pot` and apply the template to all `*po` files, update line numbers, comment out old keys, add new keys where necessary
3. Manually add translations for new keys in `*po` files. This should be the last step.

As a result, I think it is part of the process the the `*po` files should be manually updated last and thus it would not make sense to have them part of the dependencies, as they are the final product.

Reply:
I will mark this as resolved, feel free to undo it if we should discuss this further.